beets/library.py

```diff
@@ -1105,7 +1105,9 @@ def parse_query_parts(parts, model_cls):
     special path query detection.
     """
     # Get query types and their prefix characters.
-    prefixes = {':': dbcore.query.RegexpQuery}
+    prefixes = {':': dbcore.query.RegexpQuery,
+                # TODO: decide on a better symbol, or other syntax
+                u'\u00ac': dbcore.query.NotQuery}
```

Comment:
The decision of implementing the syntax via a prefix and the `¬` symbol is mainly a matter of personal choice - again, if there is a consensus on the syntax, I'd be more than happy to adapt the code to it.

The only drawback I can see about implementing is as a prefix is that it prevents the user from using negation and regular expressions together (one could argue that, if the user is using a regular expression, he could write the negation into the regular expression directly). Other than that, seems like the choice that requires less modifications on the existing code.